### ISSUE TITLE
Add Zarr Reader(s)

### ISSUE THREAD

**Issue by norlandrhagen** [zarr-developers/VirtualiZarr#262]
Opening up a tracking issue for Zarr V2 and Zarr V3/Icechunk compatible readers. It might be useful to open up multiple existing Zarr stores and virtualizarr'ize into a single Zarr. I think @maxrjones raised this possibility a while ago, but I can't find seem to find the issue. 

From what I understand from @TomNicholas the [open_virtual_dataset_from_v3_store](https://github.com/zarr-developers/VirtualiZarr/blob/7053bc02677b31177c76904d6c90a594cff30598/virtualizarr/readers/zarr.py#L16) reader is more for chunk manifest / manifest.json style v3 stores. 

- [ ]  Zarr V2 compatible reader
- [ ]  Zarr V3 compatible reader

**Comment by TomNicholas** [comment]
> the open_virtual_dataset_from_v3_store reader is more for chunk manifest / manifest.json style v3 stores.

That's correct - I think it's arguably made obsolete by the open-sourcing of Icechunk... (cc @ldeakin)

> Zarr V3/Icechunk compatible readers

Icechunk uses a different storage layout on disk, so the V3 reader won't work on it. And the Zarr store doesn't define an interface for returning virtual refs. So pulling virtual refs out of Icechunk is a separable issue, and would require Icechunk to create API to support it.

**Comment by TomNicholas** [comment]
@norlandrhagen FYI we can use https://github.com/zarr-developers/zarr-python/pull/2426 instead of `fsspec.ls`!!

**Comment by norlandrhagen** [comment]
Oh great find!

**Comment by TomNicholas** [comment]
To support generating virtual references to sharded v3 data (which we should definitely leave for a later PR), see this https://github.com/zarr-developers/zarr-python/pull/1661#issuecomment-2434419316.
